### PR TITLE
add method to create users with initial groups

### PR DESCRIPTION
**Pull request by daniel-garcia**
This MR adds the ability to create new users with initial groups. Please feel free to decline with feedback and I'll rewrite accordingly.

### CODE REVIEW

**Comment by bretterer** [comment]
Keys in travis are correct, but due to security, any PR that originates outside of the Okta org will not have access to these keys.  I will take a look at this PR.

Thank you for your contribution.  

In order to continue however, we need to make sure you submit a CLA.  Please view the details at https://developer.okta.com/cla/

**Comment by bretterer** [comment]
Another concern here is that the file edited is autogenerated from our spec.  You would need to make the changes to the swagger [templates](https://github.com/okta/okta-sdk-golang/tree/master/openapi/generator/templates) and then run the generation.

**Comment by daniel-garcia** [comment]
Taking a deeper look it appears as though it is the swagger YAML definition that in wrong. While the api supports this ability

https://developer.okta.com/docs/api/resources/users#create-user-in-group

The swagger spec, however, uses the same definition User object definition for object creation as it does for reading:
 
https://github.com/okta/okta-sdk-java/blob/master/src/swagger/api.yaml#L1593
and 
https://github.com/okta/okta-sdk-java/blob/master/src/swagger/api.yaml#L1669

, respectively. The User model is defined without the groupIds attribute. 

https://github.com/okta/okta-sdk-java/blob/master/src/swagger/api.yaml#L5135

Long story short... the problem is that there is not a separate definition for UserCreateOptions in the create call vs just User definition for other cases. This looks like a general problem with the modeling done on the swagger side (creation object differs from list/read/update object). With this in mind, I think I'll close this MR.
